jena-arq/src/test/java/org/apache/jena/sparql/function/scripting/TestScriptFunction.java

```diff
@@ -16,183 +16,222 @@
+
+import static org.junit.Assert.*;
+
+@RunWith(Parameterized.class)
+public class TestScriptFunction {
+    private static Context ctx = ARQ.getContext();
+
+    private String language;
+    private String library;
+    private String functions;
 
-public class TestJavaScriptFunctions {
-    static final String JS_LIB_FILE = "testing/ARQ/JS/test-library.js";
-    
-    static final String CamelCaseJS = 
-        StrUtils.strjoinNL
-         ("function toCamelCase(str) { return str.split(' ').map(cc).join('');}"
-         ,"function ucFirst(word)    { return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();}"
-         ,"function lcFirst(word)    { return word.toLowerCase(); }"
-         ,"function cc(word,index)   { return (index == 0) ? lcFirst(word) : ucFirst(word); }"
-         );
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                { "js", "testing/ARQ/Scripting/test-library.js",
+                        "function toCamelCase(str) { return str.split(' ').map(cc).join('');}\n"
+                        + "function ucFirst(word)    { return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();}\n"
+                        + "function lcFirst(word)    { return word.toLowerCase(); }\n"
+                        + "function cc(word,index)   { return (index == 0) ? lcFirst(word) : ucFirst(word); }\n" }
+                , {"python", "testing/ARQ/Scripting/test-library.py",
+                        "def toCamelCase(str):\n" +
+                        "  return ''.join([cc(word, index) for index, word in enumerate(str.split(' '))])\n" +
+                        "def ucFirst(word):\n" +
+                        "  return word[0].upper() + word[1:].lower()\n" +
+                        "def lcFirst(word):\n" +
+                        "  return word.lower()\n" +
+                        "def cc(word,index):\n" +
+                        "  if index == 0:\n" +
+                        "    return lcFirst(word)\n" +
+                        "  return ucFirst(word)\n" }
+        });
+    }
+
+
+    public TestScriptFunction(String language, String library, String functions) {
+        this.language = language;
+        this.library = library;
+        this.functions = functions;
+    }
 
-    private static EnvJavaScript setupJS() {
-        Context cxt = ARQ.getContext().copy();
-        cxt.set(EnvJavaScript.symJavaScriptFunctions, CamelCaseJS);
-        cxt.set(EnvJavaScript.symJavaScriptLibFile, JS_LIB_FILE);
-        return EnvJavaScript.create(cxt);
+    @Before
+    public void setup() {
+        ctx.set(LanguageSymbols.scriptLibrary(language), library);
+        ctx.set(LanguageSymbols.scriptFunctions(language), functions);
     }
 
-    private EnvJavaScript envJS = setupJS();
+    @After
+    public void teardown() {
+        ctx.unset(LanguageSymbols.scriptFunctions(language));
+        ctx.unset(LanguageSymbols.scriptFunctions(language));
```

Comment:
Shouldn't one of these two calls be unsetting the scriptLibrary?

Reply:
good catch.